internal/app/common/continuous_tester.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/messenger"
+)
+
+type (
+	IContinuousTester interface {
+		// Start test. Blocks until error.
+		Start(start func(ctx context.Context) (int, error), testDuration, pauseBetweenTests time.Duration) error
+		Cancel()
+		Done() <-chan struct{}
+	}
+
+	ContinuousTesterOptions struct {
+		PagerDutyIntegrationKey string
+		PagerDutyComponent      string
+		PagerDutyLowUrgency     bool
+		TesterOptions
+	}
+
+	continuousTester struct {
+		ctx                     context.Context
+		cancel                  context.CancelFunc
+		host                    string // API host being tested
+		pagerDutyIntegrationKey string
+		pagerDutyComponent      string
+		pagerDutyLowUrgency     bool
+		name                    string
+	}
+)
+
+func NewContinuousTester(gctx context.Context, opts ContinuousTesterOptions, testerName string) IContinuousTester {
+	ctx, cancel := context.WithCancel(gctx)
+	server := opts.API.GetServer()
+	u, _ := url.Parse(server)
+	ct := &continuousTester{
+		ctx:                     ctx,
+		cancel:                  cancel,
+		host:                    u.Host,
+		pagerDutyIntegrationKey: opts.PagerDutyIntegrationKey,
+		pagerDutyComponent:      opts.PagerDutyComponent,
+		pagerDutyLowUrgency:     opts.PagerDutyLowUrgency,
+		name:                    testerName,
+	}
+	return ct
+}
+
+func (ct *continuousTester) Start(start func(ctx context.Context) (int, error), testDuration, pauseBetweenTests time.Duration) error {
+	messenger.SendMessage(fmt.Sprintf("Starting continuous %s test of %s", ct.name, ct.host))
+	for {
+		msg := fmt.Sprintf(":arrow_right: Starting %s %s test to %s", testDuration, ct.name, ct.host)
+		messenger.SendMessage(msg)
+
+		ctx, cancel := context.WithTimeout(ct.ctx, testDuration)
+		es, err := start(ctx)
+		ctxErr := ctx.Err()
+		cancel()
+
+		if ct.ctx.Err() != nil {
+			messenger.SendMessage(fmt.Sprintf("Continuous test of %s on %s cancelled", ct.name, ct.host))
+			return ct.ctx.Err()
+		} else if ctxErr != nil {
+			msg := fmt.Sprintf("Test of %s on %s timed out, potential deadlock! ctxErr=%q err=%q", ct.name, ct.host, ctxErr, err)
+			messenger.SendFatalMessage(msg)
+		} else if err != nil || es != 0 {
+			msg := fmt.Sprintf(":rotating_light: Test of %s on %s ended with err=%v errCode=%v", ct.name, ct.host, err, es)
+			messenger.SendFatalMessage(msg)
+			glog.Warning(msg)
+			ct.sendPagerdutyEvent(err)
```

Comment:
I know you're probably following existing patterns in stream-tester, but I've been really trying to drive everything toward Grafana. Centralizing our alerting there  allows everything to be managed in one place, allows alerts to be tuned and silenced when necessary — very useful.

Reply:
I think centralizing everything with Grafana is a good idea. Though, I think we should change it in a separate PR. What do you think?

This PR is just to add the Transcode API support in the exactly the same way we support VOD API.

Btw. do you have any example of using Grafana in streamtester?

Reply:
Ok, merging this PR. We can do the refactoring to use Grafana (for both VOD and Storj) in a separate PR.